src/CommitmentsV2/SFA.DAS.CommitmentsV2/Application/Commands/OverlappingTrainingDateRequestAutomaticStopAfter2Weeks/OverlappingTrainingDateRequestAutomaticStopAfter2WeeksHandler.cs

```diff
@@ -111,6 +115,17 @@ private async Task AutoStopOverlappingTrainingDateRequest(OverlappingTrainingDat
                 Party.Employer));
         }
 
+        private static DateTime SetStopDateIfDraftApprenticeshipHasFutureStartDate(OverlappingTrainingDateRequest request, DateTime stopDate, DateTime today)
+        {
+            if (request.PreviousApprenticeship.StartDate.Value < today
+                && request.PreviousApprenticeship.EndDate.Value > today
+                && request.DraftApprenticeship.StartDate.Value > today)
+            {
```

Comment:
just curious can this logic work? Can the StartDate be both before today and after today, it appears to be a slight contradiction.

Also is it possible to enter a stop date before the the course has started (I recall some rules around this, which would be good to review)? Be good to discuss this with Najam, he's the oracle for me

Reply:
The startDate of the PreviousApprenticeship is in the past and the StartDate of the DraftApprenticeship is in the future. This is possible.

Reply:
It is possible to set a stop date in the future if the Apprenticeship in question is WaitingToStart

Reply:
apologies, I missed StartDate was from Previous and Draft apprenticeships. (I blame my age)